crates/swc_common/src/plugin.rs

```diff
@@ -8,38 +8,64 @@ use crate::{syntax_pos::Mark, SyntaxContext};
 use anyhow::Error;
 use std::any::type_name;
 
-/// Serialize ast to pass into plugins. This is being called from host side.
-/// TODO: https://github.com/swc-project/swc/issues/3167
-#[cfg(feature = "plugin-base")]
-pub fn serialize_for_plugin<T>(t: &T) -> Result<rkyv::AlignedVec, Error>
-where
-    T: rkyv::Serialize<rkyv::ser::serializers::AllocSerializer<512>>,
-{
-    rkyv::to_bytes::<_, 512>(t).map_err(|err| match err {
-        rkyv::ser::serializers::CompositeSerializerError::SerializerError(e) => e.into(),
-        rkyv::ser::serializers::CompositeSerializerError::ScratchSpaceError(e) => {
-            Error::msg("AllocScratchError")
-        }
-        rkyv::ser::serializers::CompositeSerializerError::SharedError(e) => {
-            Error::msg("SharedSerializeMapError")
-        }
-    })
+/// Wraps internal representation of serialized data. Consumers should not
+/// rely on specific details of byte format struct contains: it is
+/// strictly implementation detail which can change anytime.
+pub struct Serialized<T> {
+    field: T,
```

Comment:
I think `Serialized` should have `AlignedVec` with `PhantomData<T>`.
This seems a bit weird as `T` is not serialized.

Reply:
if we're going to use rkyv for any kind of serialization, that may works better. I thought we may have some other cases insetad.

Reply:
I'm not sure about it at the moment, but if we want to store alternative `Serialized` in `swc_common`, we can make a breaking change.

But I think we can use `rkyv` for almost everything as `rkyv` is zero copy serialization framework. It's faster than `memmove` which is required for `abi_stable`

Reply:
yes, makes sense. Probably then we don't need to have generic for the wrapping struct, as long as we can hide alignedvec from consumer.